CC-12884 support configurable products in the wishlist (#594)
spryker/spryker-docs: docs/scos/dev/glue-api-guides/202200.0/managing-wishlists/managing-wishlist-items.md

                "prices": [
                    {
                        "priceTypeName": "DEFAULT",
                        "netAmount": 23434,
                        "grossAmount": 42502,
                        "currency": {
                            "code": "EUR",
                            "name": "Euro",
                            "symbol": "€"
                        },
                        "volumePrices": [
                            {
                                "netAmount": 150,
                                "grossAmount": 165,
                                "quantity": 5
                            },
                            {
                                "netAmount": 145,
                                "grossAmount": 158,
                                "quantity": 10
                            },
                            {
                                "netAmount": 140,
                                "grossAmount": 152,
                                "quantity": 20
                            }
                        ]
                    }
                ]
            }
        }
    }
}
```

</details>

> <a name="request-attributes-description"></a>

Reviewer: Is it normal to have a link without an HREF tag?

Author: this is an anchor to which I'm referring to from another place in the document

                        "currency": {
                            "code": "EUR",
                            "name": "Euro",
                            "symbol": "€"
                        },
                        "volumePrices": [
                            {
                                "grossAmount": 165,
                                "netAmount": 150,
                                "quantity": 5
                            },
                            {
                                "grossAmount": 158,
                                "netAmount": 145,
                                "quantity": 10
                            },
                            {
                                "grossAmount": 152,
                                "netAmount": 140,
                                "quantity": 20
                            }
                        ]
                    }
                ]
            },
            "prices": []
        },
        "links": {
            "self": "http://glue.de.spryker.local/wishlists/b70e1073-a740-5a48-bb5e-0449a9e51d53/wishlist-items/093_24495843_98bf36f052d23f10a8a081694ad4f45e"
        }
    }
}
```

</details>

> <a name="wishlishlist-items-response-attributes"></a>

Reviewer: Is it normal to have a link without an HREF tag?

Author: this is an anchor to which I'm referring to from another place in the document

| displayData  | Array  | Array of variables a Storefront user set up in the configurator.  |
| configuration  | Array  | Default product configuration.  |
| configuratorKey  | String  | Configurator type.  |
| isComplete  | Boolean  | Shows if the configurable product configuration is complete:<div><ul><li>`true`—configuration complete.</li><li>`false`—configuration incomplete.</li></ul></div>  |
| quantity  | Integer  | Quantity of the product that was added to the wishlist.  |
| availableQuantity  | Integer  | Product quantity available in the store. |

For attribute descriptions of product prices, see [Retrieving abstract product prices](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-products/abstract-products/retrieving-abstract-product-prices.html#abstract-product-prices-response-attributes).

For attribute desriptions of concrete product availabillity, see [Retrieving concrete product availability](/docs/scos/dev/glue-api-guides/202108.0/managing-products/concrete-products/retrieving-concrete-product-availability.html#concrete-product-availability-response-attributes).

## Delete a wishlist item

To delete wishlist item, send the request:

---

> `DELETE` **/wishlists/*{% raw %}{{{% endraw %}wishlist_id{% raw %}}}{% endraw %}*/wishlist-items/*{% raw %}{{{% endraw %}item_sku{% raw %}}}{% endraw %}***

Reviewer 2: I think will be better to use `wishlist_item_id` instead of `item_sku` in request description.
Especially since after refactoring we replaced `sku` with `uuid`

Author: done

---

| PATH PARAMETER | DESCRIPTION |
| --- | --- |
| ***{% raw %}{{{% endraw %}wishlist_id{% raw %}}}{% endraw %}*** | Unique identifier of the wishlist to delete an item from. [Create a wishlist](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-wishlists/managing-wishlists.html#create-a-wishlist) or [retrieve all wishlists](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-wishlists/managing-wishlists.html#retrieve-wishlists) to get it. |
| ***{% raw %}{{{% endraw %}item_sku{% raw %}}}{% endraw %}*** | Unique identifier of the product to delete. |

### Request

| HEADER KEY | HEADER VALUE | REQUIRED | DESCRIPTION |
| --- | --- | --- | --- |
| Authorization | string | &check; | Alphanumeric string that authorizes the customer to send requests to protected resources. Get it by [authenticating as a customer](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-customers/authenticating-as-a-customer.html).  |

> Request sample: `DELETE https://glue.mysprykershop.com/wishlists/09264b7f-1894-58ed-81f4-d52d683e910a/wishlist-items/064_18404924`

Reviewer 2: I think we should describe additional request with a configurable product in case.
I this case we use not just `sku`  but `sku_productConfigurationInstanceHash` to identify item with the same sku but different configuration. Schema of usage the same. Recive this identifer we can also when adding wishlist item or getting items `"id": "093_24495843_98bf36f052d23f10a8a081694ad4f45e"`

Author: So, basically, what I should do is to add an additional `DELELE` request, but with `sku_productConfigurationInstanceHash` instead of `wishlist_item_id`, right?

Author: Done.

### Response

If the item is removed successfully, the endpoint returns the `204 No Content` status code.

## Possible errors

| CODE | REASON |
| --- | --- |
| 201 | Cannot find the wishlist. |
| 206 | Cannot add an item to the wishlist. |
| 207 | Cannot remove the item. |
| 208 | An item with the provided SKU does not exist in the wishlist. |

> To view generic errors that originate from the Glue Application, see [Reference information: GlueApplication errors](/docs/scos/dev/glue-api-guides/{{page.version}}/reference-information-glueapplication-errors.html).

Reviewer 2: Also we can update wishlist item by `path: "/wishlists/{wishlistId}/wishlist-items/{wishlistItemId}"`
```
"parameters": [{
                  "ref": "acceptLanguage"
              }],
"responses": {
                  "403": "Unauthorized request.",
                  "404": "Wishlist not found.",
                  "422": "Cannot update the wishlist list item."
 },
```

| isComplete  | Boolean  | Shows if the configurable product configuration is complete:<div><ul><li>`true`—configuration complete.</li><li>`false`—configuration incomplete.</li></ul></div>  |
| quantity  | Integer  | Quantity of the product that is added to the wishlist.  |
| availableQuantity  | Integer  |  Product quantity available in the store. |

For attribute descriptions of product prices, see [Retrieving abstract product prices](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-products/abstract-products/retrieving-abstract-product-prices.html#abstract-product-prices-response-attributes).

### Response

Resonse sample:

```json
{
		"data": {
			"type": "wishlist-items",
			"id": "064_18404924",
			"attributes": {
				"sku": "064_18404924"
			},
			"links": {
				"self": "https://glue.mysprykershop.com/wishlists/c917e65b-e8c3-5c8b-bec6-892529c64b30/wishlist-items/064_18404924"
			}
		}
	}
```

<details><summary>Response sample: add a configurable product to a wishlist</summary>

```json
{
    "data": {
        "type": "wishlist-items",
        "id": "093_24495843_98bf36f052d23f10a8a081694ad4f45e",
        "attributes": {
            "productOfferReference": null,

>             "merchantReference": "MER000001",

Author: these attributes are related to Marketplace indeed, and they are always returned in the responce. Added respective descriptions.

| --- | --- |
| ***{% raw %}{{{% endraw %}wishlist_id{% raw %}}}{% endraw %}*** | Unique identifier of the wishlist to delete an item from. [Create a wishlist](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-wishlists/managing-wishlists.html#create-a-wishlist) or [retrieve all wishlists](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-wishlists/managing-wishlists.html#retrieve-wishlists) to get it. |
| ***{% raw %}{{{% endraw %}wishlist_item_id{% raw %}}}{% endraw %}*** | Unique identifier of a configurable product to update. To get this identifier, [retrieve a wishlist with items included](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-wishlists/managing-wishlists.html#retrieve-wishlists) or [add a configurable product to a wishlist](#add-an-item-to-a-wishlist). |

### Request

| HEADER KEY | HEADER VALUE | REQUIRED | DESCRIPTION |
| --- | --- | --- | --- |
| Authorization | string | &check; | Alphanumeric string that authorizes the customer to send requests to protected resources. Get it by [authenticating as a customer](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-customers/authenticating-as-a-customer.html).  |

> <details><summary>Request sample: add a configurable product to a wishlist</summary>

Reviewer 3: ```suggestion
<details><summary>Request sample: Update a configurable product in a wishlist</summary>
```

                                "quantity": 20
                            }
                        ]
                    }
                ]
            }
        }
    }
}
```

</details>

| ATTRIBUTE | TYPE | DESCRIPTION |
| --- | --- | --- |
| sku | String | Configurable product SKU to update. |
| displayData  | Array  | Array of variables that are proposed for a Storefront user to set up in the configurator.  |
| configuration  | Array  | Default configurable product configuration.  |
| configuratorKey  | String  | Configurator type.  |
| isComplete  | Boolean  | Shows if the configurable product configuration is complete:<div><ul><li>`true`—configuration complete.</li><li>`false`—configuration incomplete.</li></ul></div>  |
| quantity  | Integer  | Quantity of the configurable product to update in the wishlist.  |
| availableQuantity  | Integer  |  Product quantity available in the store. |

For attribute descriptions of product prices, see [Retrieving abstract product prices](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-products/abstract-products/retrieving-abstract-product-prices.html#abstract-product-prices-response-attributes).

### Response

> <details><summary>Response sample: add a configurable product to a wishlist</summary>

Reviewer 3: ```suggestion
<details><summary>Response sample: Update a configurable product in a wishlist</summary>
```

                                "netAmount": 140,
                                "quantity": 20
                            }
                        ]
                    }
                ]
            },
            "prices": []
        },
        "links": {
            "self": "https://glue.mysprykershop.com/wishlists/b70e1073-a740-5a48-bb5e-0449a9e51d53/wishlist-items/093_24495843_94a23d655bf161d6ab0088457f4ea2fc"
        }
    }
}
```

</details>

| ATTRIBUTE | TYPE | DESCRIPTION |
| --- | --- | --- |
| sku | String | Updated configurable product SKU. |
| displayData  | Array  | Array of variables a Storefront user set up in the configurator.  |
| configuration  | Array  | Updated configurable product configuration.  |
| configuratorKey  | String  | Configurator type.  |
| isComplete  | Boolean  | Shows if the configurable product configuration is complete:<div><ul><li>`true`—configuration complete.</li><li>`false`—configuration incomplete.</li></ul></div>  |
| quantity  | Integer  | Updated configurable product quantity in the wishlist.  |
| availableQuantity  | Integer  | Configurable product quantity available in the store. |

> For attribute descriptions of product prices, see [Retrieving abstract product prices](/docs/scos/dev/glue-api-guides/{{page.version}}/managing-products/abstract-products/retrieving-abstract-product-prices.html#abstract-product-prices-response-attributes).

Reviewer 3: please see above for the correct link to prices attributes

Author: Done
